controller/setting_controller.go

```diff
@@ -1062,3 +1073,350 @@ func (sc *SettingController) cleanupFailedSupportBundles() error {
+				logrus.WithError(err).Debugf("Failed to get %v Pod", pod.Name)
+				continue
+			}
+			for _, container := range podMetrics.Containers {
+				totalCPUUsage.Add(*container.Usage.Cpu())
+				totalMemoryUsage.Add(*container.Usage.Memory())
+			}
+		}
+
+		avgCPUUsageMilli := totalCPUUsage.MilliValue() / int64(len(pods))
+		avgCPUUsage := resource.NewMilliQuantity(avgCPUUsageMilli, resource.DecimalSI).String()
+		cpuStruct := util.StructName(fmt.Sprintf(ExtraInfoPodAvgCPUUsageFmt, component))
+		info.structFields.Append(cpuStruct, avgCPUUsage, info.structFields.ConvertTag(cpuStruct))
+
+		avgMemoryUsageBytes := totalMemoryUsage.Value() / int64(len(pods))
+		avgMemoryUsageMiB := float64(avgMemoryUsageBytes) / (1024 * 1024)
+		avgMemoryUsage := resource.NewQuantity(int64(avgMemoryUsageMiB), resource.BinarySI).String()
+		memStruct := util.StructName(fmt.Sprintf(ExtraInfoPodAvgMemoryUsageFmt, component))
+		info.structFields.Append(memStruct, avgMemoryUsage, info.structFields.ConvertTag(memStruct))
+	}
+
+	return nil
+}
+
+func (info *ExtraInfo) collectSettings() error {
+	excludes := map[types.SettingName]bool{
+		types.SettingNameBackupTargetCredentialSecret:      true,
+		types.SettingNameDefaultEngineImage:                true,
+		types.SettingNameDefaultInstanceManagerImage:       true,
+		types.SettingNameDefaultBackingImageManagerImage:   true,
+		types.SettingNameSupportBundleManagerImage:         true,
+		types.SettingNameCurrentLonghornVersion:            true,
+		types.SettingNameLatestLonghornVersion:             true,
+		types.SettingNameStableLonghornVersions:            true,
+		types.SettingNameDefaultLonghornStaticStorageClass: true,
+		types.SettingNameDeletingConfirmationFlag:          true,
+		types.SettingNameDefaultDataPath:                   true,
+		types.SettingNameUpgradeChecker:                    true,
+	}
+
+	// Settings that require extra processing to identify their general purpose
+	includeAsCustom := map[types.SettingName]bool{
+		types.SettingNameBackupTarget: true, // get the backup target type, ex: cifs, nfs, s3
+	}
+
+	// Settings that should be collected as boolean (true if configured, false if not)
+	includeAsBoolean := map[types.SettingName]bool{
+		types.SettingNameTaintToleration:                     true,
+		types.SettingNameSystemManagedComponentsNodeSelector: true,
+		types.SettingNameRegistrySecret:                      true,
+		types.SettingNamePriorityClass:                       true,
+		types.SettingNameStorageNetwork:                      true,
+	}
+
+	settings, err := info.ds.ListSettings()
+	if err != nil {
+		return err
+	}
+
+	settingMap := make(map[string]string)
+	for _, setting := range settings {
+		settingName := types.SettingName(setting.Name)
+
+		switch {
+		case excludes[settingName]:
+			continue
+		case includeAsBoolean[settingName]:
+			settingMap[setting.Name] = fmt.Sprint(setting.Value != "")
+		case includeAsCustom[settingName] && settingName == types.SettingNameBackupTarget:
```

Comment:
Why not just `case settingName == types.SettingNameBackupTarget`? no need to strictly follow the group pattern (I know you try to group settings for different types)